src/main/java/io/stargate/sgv2/jsonapi/api/model/command/builders/CollectionSortClauseBuilder.java

```diff
@@ -12,6 +21,107 @@ public CollectionSortClauseBuilder(CollectionSchemaObject collection) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' no other sort expressions allowed",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD);
+      }
+      if (!lexicalValue.isTextual()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "if sorting by '%s' value must be String, not %s",
+            DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
+            JsonUtil.nodeTypeAsString(lexicalValue));
+      }
+
+      return new SortClause(
+          Collections.singletonList(
+              SortExpression.collectionLexicalSort(lexicalValue.textValue())));
+    }
+
+    // Otherwise, use shared default processing
     return defaultBuildAndValidate(sortNode);
   }
+
+  @Override
+  protected String validateSortClausePath(String path) {
+    if (!NamingRules.FIELD.apply(path)) {
+      if (path.isEmpty()) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
+            "path must be represented as a non-empty string");
+      }
+      // But allow "well-known" fields
+      switch (path) {
+        case DocumentConstants.Fields.LEXICAL_CONTENT_FIELD,
+            DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD,
+            DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD -> {}
+        default ->
+            throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
+                "path ('%s') cannot start with '$' (except for pseudo-fields '$lexical', '$vector' and '$vectorize')",
+                path);
+      }
+    }
+
+    try {
+      path = DocumentPath.verifyEncodedPath(path);
+    } catch (IllegalArgumentException e) {
+      throw ErrorCodeV1.INVALID_SORT_CLAUSE_PATH.toApiException(
+          "sort clause path ('%s') is not a valid path. " + e.getMessage(), path);
+    }
+
+    return path;
+  }
+
+  @Override
+  protected SortExpression buildAndValidateExpression(
+      String validatedPath, JsonNode innerValue, int totalFields) {
+    if (DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(validatedPath)) {
+      // Vector search can't be used with other sort clause
+      if (totalFields > 1) {
+        throw ErrorCodeV1.VECTOR_SEARCH_USAGE_ERROR.toApiException();
+      }
+      float[] vectorFloats = tryDecodeBinaryVector(validatedPath, innerValue);
+      if (vectorFloats == null) {
+        if (!innerValue.isArray()) {
+          throw ErrorCodeV1.SHRED_BAD_VECTOR_VALUE.toApiException();
+        }
+        ArrayNode arrayNode = (ArrayNode) innerValue;
+        vectorFloats = new float[arrayNode.size()];
+        if (arrayNode.isEmpty()) {
+          throw ErrorCodeV1.SHRED_BAD_VECTOR_SIZE.toApiException();
+        }
+        for (int i = 0; i < arrayNode.size(); i++) {
+          JsonNode element = arrayNode.get(i);
+          if (!element.isNumber()) {
+            throw ErrorCodeV1.SHRED_BAD_VECTOR_VALUE.toApiException();
+          }
+          vectorFloats[i] = element.floatValue();
```

Comment:
do we have re-usable code in shredding for moving an array into float[]?

the table builder calls arrayNodeToVector()

Reply:
Good catch, changed.